unittests/test_utility.py

```diff
@@ -1004,3 +1004,23 @@ def test_concat_files(self):
                 with open(concat_file) as cf:
                     out = cf.read()
                     assert out == 'Hello1\nHello2\n'
+
+    def test_unique_abs_paths(self):
+        p1 = 'a/b/c'
+        p1_copy = p1[:]
+        p1_parent = 'a/b'
+        p2 = '/d/e//'
+        p3 = '/d/e/f'
```

Comment:
It's not easy to follow what the unit tests are doing when you are using these variables for the paths. Just hard code them as list arguments. A bit of duplication here serves much more here, because the tests will be self-explaining.